apps/web/src/design-system/sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,156 @@
+import { ReactNode, useEffect } from 'react';
+import { ActionIcon, createStyles, Drawer, Loader, MantineTheme, Stack } from '@mantine/core';
+import styled from '@emotion/styled';
+
+import { colors, shadows } from '../config';
+import { ArrowLeft, Close } from '../icons';
+import { When } from '../../components/utils/When';
+
+const HeaderHolder = styled.div`
+  display: flex;
+  flex-wrap: nowrap;
+  gap: 12px;
+  margin: 24px;
+  margin-bottom: 0;
+`;
+
+const BodyHolder = styled.div`
+  display: flex;
+  flex-direction: column;
+  overflow-x: hidden;
+  overflow-y: auto;
+  margin: 0 24px;
+  gap: 24px;
+`;
+
+const FooterHolder = styled.div`
+  display: flex;
+  flex-wrap: nowrap;
+  gap: 6px;
+  margin: 24px;
+  margin-top: 0;
+  margin-top: auto;
+`;
+
+const COLLAPSED_WIDTH = 480;
+const HEADER_HEIGHT = 65;
+const NAVIGATION_WIDTH = 300;
+const PAGE_MARGIN = 30;
+const INTEGRATION_SETTING_TOP = HEADER_HEIGHT + PAGE_MARGIN;
+
+const useDrawerStyles = createStyles((theme: MantineTheme) => {
+  return {
+    root: {
+      position: 'absolute',
+      zIndex: 1,
+    },
+    drawer: {
+      position: 'fixed',
+      top: `${INTEGRATION_SETTING_TOP}px`,
+      right: `${PAGE_MARGIN}px`,
+      bottom: 0,
+      backgroundColor: theme.colorScheme === 'dark' ? colors.B17 : colors.white,
+      borderRadius: 7,
+      boxShadow: shadows.dark,
+    },
+    body: {
+      height: '100%',
+    },
+  };
+});
```

Comment:
the sidebar styles... I feel it's not perfect because of this gap, asked Nikolay about his thoughts and we might fix that in the next iteration...
![Screenshot 2023-07-07 at 12 20 37](https://github.com/novuhq/novu/assets/2607232/43d410fc-e550-49f0-bf92-837b4b32d6f7)